client/src/components/UI/BasePeriodSelect.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+    <v-select v-model="period" :items="values" item-text="text" item-value="value" label="Time Period" />
+</template>
+
+<script>
+import { mapActions } from 'vuex';
+
+export default {
+    data() {
+        return {
+            period: 1,
+            values: [
+                { text: '1st week of January', value: 1 },
+                { text: '2nd week of January', value: 2 },
+                { text: '3rd week of January', value: 3 },
+                { text: '4th week of January', value: 4 },
+                { text: '5th week of January', value: 5 }
+            ]
+        };
+    },
+
+    watch: {
+        period() {
+            this.fetchTrafficByTime();
+        }
+    },
+
+    methods: {
+        ...mapActions({ fetchTrafficEntriesByTime: 'traffic/fetchEtriesByTime' }),
+
+        async fetchTrafficByTime() {
+            let between;
+
+            switch (this.period) {
+                case 1:
+                    between = {
+                        from: '2021-01-01',
+                        to: '2021-01-07'
+                    };
+                    break;
+                case 2:
+                    between = {
+                        from: '2021-01-08',
+                        to: '2021-01-14'
+                    };
+                    break;
+                case 3:
+                    between = {
+                        from: '2021-01-15',
+                        to: '2021-01-21'
+                    };
+                    break;
+                case 4:
+                    between = {
+                        from: '2021-01-22',
+                        to: '2021-01-28'
+                    };
+                    break;
+                case 5:
+                    between = {
+                        from: '2021-01-29',
+                        to: '2021-01-31'
+                    };
+                    break;
+            }
+
+            await this.fetchTrafficEntriesByTime(between);
+        }
+    },
+
+    created() {
```

Comment:
please always install eslint, created above methods

Reply:
I have istalled eslint and it didn't throw any error. But I'm gonna change it. My .eslintrc.js is:
`module.exports = {
    root: true,
    env: {
        node: true
    },
    extends: ['plugin:vue/essential', 'eslint:recommended', '@vue/prettier'],
    parserOptions: {
        parser: 'babel-eslint'
    },
    rules: {
        'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
        'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
    },
    overrides: [
        {
            files: ['**/__tests__/*.{j,t}s?(x)', '**/tests/unit/**/*.spec.{j,t}s?(x)'],
            env: {
                jest: true
            }
        }
    ]
};
`

Reply:
if you want to paste some longer code then use 3x tilde
[3x tildes]
code
[3x tildes]

and about errors, have you checked console for warning?

Reply:
Well... I don't remember. I'm gonna check console for warnings.